sqlglot/dialects/clickhouse.py

```diff
@@ -1106,3 +1106,9 @@ def replacepartition_sql(self, expression: exp.ReplacePartition) -> str:
 
         def projectiondef_sql(self, expression: exp.ProjectionDef) -> str:
             return f"PROJECTION {self.sql(expression.this)} {self.wrap(expression.expression)}"
+
+        def except_op(self, expression: exp.Except) -> str:
+            return f"EXCEPT{' DISTINCT' if expression.args.get('distinct') else ' ALL'}"
+
+        def intersect_op(self, expression: exp.Intersect) -> str:
+            return f"INTERSECT{' DISTINCT' if expression.args.get('distinct') else ' ALL'}"
```

Comment:
What about `union_op`? I see that we have a `EXPLICIT_SET_OP` flag, maybe we can just fix the base generator methods and set this flag to true for clickhouse?

Reply:
It's already set to `True`. Let me see how weird things get with working on the base op.

Reply:
Ah, nice. Yeah, I meant fiddling with the base methods and see if that allows us to avoid overriding them.

Reply:
Ok, not too bad hopefully!